### PR TITLE
Completed Assessment Api untested

### PR DESCRIPTION
**Pull request by senseihimanshu**

### CODE REVIEW

File: arms-services/src/Arms.Api/Controllers/AssessmentController.cs [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+            this._assessmentRepository = assessmentRepository;
+        }
+
+        [HttpGet("")]
+        private JsonResult Index()
+        {
+            var model = _assessmentRepository.GetAllAssessments();
+            return Json(model);
+        }
+
```
Comment (sonalichawla16) [inline arms-services/src/Arms.Api/Controllers/AssessmentController.cs:23]: Correct routes

File: arms-services/src/Arms.Api/Controllers/AssessmentController.cs [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+            return Json(model);
+        }
+        
+        [HttpPut("")]
+        private JsonResult Modify(Assessment assessment)
+        {
+            var model = _assessmentRepository.Update(assessment);
+            return Json(model);
+        }
+
```
Comment (sonalichawla16) [inline arms-services/src/Arms.Api/Controllers/AssessmentController.cs:44]: Not a standard JSon result
no status
No payload
No message

File: arms-services/src/Arms.Api/Models/AssessmentRepository.cs [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+            _context.Assessments.Add(assessment);
+            _context.SaveChanges();
+            return assessment;
+        }
+
+        public Assessment Update(Assessment assessmentChanges)
+        {
+            var employee = _context.Assessments.Attach(assessmentChanges);
+            employee.State = Microsoft.EntityFrameworkCore.EntityState.Modified;
+            _context.SaveChanges();
```
Comment (sonalichawla16) [inline arms-services/src/Arms.Api/Models/AssessmentRepository.cs:39]: Make it a standard, which should be used in all existing APIs

File: arms-services/src/Arms.Api/Controllers/AssessmentController.cs (hunk 1 above)
Comment (senseihimanshu) [reply arms-services/src/Arms.Api/Controllers/AssessmentController.cs:42]: Corrected

File: arms-services/src/Arms.Api/Controllers/AssessmentController.cs (hunk 2 above)
Comment (senseihimanshu) [reply arms-services/src/Arms.Api/Controllers/AssessmentController.cs:125]: Corrected

File: arms-services/src/Arms.Api/Models/AssessmentRepository.cs (hunk 3 above)
Comment (senseihimanshu) [reply arms-services/src/Arms.Api/Models/AssessmentRepository.cs]: Corrected!

File: ARMS_DB_Scripts/Interview.sql [hunk 4]
```diff
@@ -27,8 +36,8 @@ AS
 
 GO
 --Insert command for interview
-Insert into ARMS.Interview([date],[time],[venue],[noOfRounds],[jobId])
-Values('2020-02-09','12:00','delhi','5','1')
+--Insert into ARMS.Interview([date],[time],[venue],[noOfRounds],[jobId])
+--Values('2020-02-09','12:00','delhi','5','1')
```
Comment (sonalichawla16) [inline ARMS_DB_Scripts/Interview.sql:41]: Why comment not delete

File: ARMS_DB_Scripts/Location.sql [hunk 5]
```diff
@@ -14,24 +20,25 @@ BEGIN
...
     WHERE id IN (SELECT DISTINCT id FROM Inserted)
-
+END
 GO
-insert into ARMS.Location ([locationName],[createdBy],[modifiedBy])
-values ('Dallas', 'shivani','shivani'),
- ('Noida', 'shivani','shivani')
+
+--insert into ARMS.Location ([locationName],[jobId],[createdBy],[modifiedBy])
+--values ('Dallas', 1, 'shivani','shivani');
```
Comment (sonalichawla16) [inline ARMS_DB_Scripts/Location.sql:42]: ....???

File: arms-services/src/Arms.Api/Controllers/AssessmentController.cs [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+            Response<IEnumerable<Assessment>> response;
+            
+            try
+            {
+                List<Assessment> data = _context.Assessment.ToList();
+                response = new Response<IEnumerable<Assessment>>(true, data, "Assessments retrieved successfully");
+            }
+            catch (Exception e)
+            {
+                Console.WriteLine(e);
```
Comment (sonalichawla16) [inline arms-services/src/Arms.Api/Controllers/AssessmentController.cs:35]: remove console logs

File: arms-services/src/Arms.Api/Controllers/AssessmentController.cs [hunk 7]
```diff
@@ -0,0 +1,163 @@
...
+            try
+            {
+                Assessment data = _context.Assessment.FirstOrDefault(assessment => assessment.Id == id);
+                response = new Response<Assessment>(true, data, "Assessment retrieved successfully");
+            }
+            catch (Exception e)
+            {
+                Console.WriteLine(e);
+                response = new Response<Assessment>(false, null, "Something went wrong");
+
```
Comment (sonalichawla16) [inline arms-services/src/Arms.Api/Controllers/AssessmentController.cs:57]: remove console logs

File: arms-services/src/Arms.Api/Controllers/AssessmentController.cs [hunk 8]
```diff
@@ -0,0 +1,163 @@
...
+                };
+                _context.Assessment.Add(data);
+                _context.SaveChanges();
+                response = new Response<Assessment>(true, data, "Assessment Created successfully");
+            }
+            catch (Exception e)
+            {
+                Console.WriteLine(e);
+                response = new Response<Assessment>(false, null, "Something went wrong");
+
```
Comment (sonalichawla16) [inline arms-services/src/Arms.Api/Controllers/AssessmentController.cs:91]: ?

File: arms-services/src/Arms.Infrastructure/EntityTypeConfigurations/ArmsEmployeeRolesEntityTypeConfiguration.cs [hunk 9]
```diff
@@ -7,43 +7,43 @@
...
+    //     //     builder.Property(e => e.DateCreated)
+    //     //         .HasColumnName("DateCreated")
+    //     //         .HasDefaultValueSql("(sysdatetime())");
+    //     //
+    //     //     builder.Property(e => e.DateModified)
+    //     //         .HasColumnName("DateModified")
+    //     //         .HasDefaultValueSql("(sysdatetime())");
+    //     //
+    //     //     builder.Property(e => e.RoleOrder)
+    //     //         .HasColumnName("RoleOrder");
```
Comment (sonalichawla16) [inline arms-services/src/Arms.Infrastructure/EntityTypeConfigurations/ArmsEmployeeRolesEntityTypeConfiguration.cs:44]: Never ever comment someone's else comment, just for the sake of fixing build issues at your end.

**Pull request merged**